analyzers/tests/SonarAnalyzer.Test/Helpers/AspNetMvcHelperTest.cs

```diff
@@ -37,6 +37,10 @@ public class Foo : System.Web.Mvc.Controller
     protected void ProtectedFoo() { }
     internal void InternalFoo() { }
     private void PrivateFoo() { }
+    public static void StaticFoo() { }
+    public void OutFoo(out string arg) { arg = null; }
+    public void RefFoo(ref string arg) { }
```

Comment:
NIT: Add the other ref kinds as well under test:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/method-parameters#reference-parameters

Reply:
These are all ordinary methods. We should also put at least one other [method kind](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.methodkind?view=roslyn-dotnet-4.7.0) under test. I think a constructor with parameters should be enough to cover the "Ordinary" check.

Reply:
Some [method modifiers](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#1561-general) are also interesting to put under test:
* new, virtual, override, sealed -> okay
* abstract -> okay
* extern -> Not okay (extern method need to be static anyway)
* unsafe -> I don't know. It shouldn't matter, so probably okay.

Reply:
The constructor test is missing. https://github.com/SonarSource/sonar-dotnet/pull/9183/files#r1579066058